### PR TITLE
Core/Commands: Add new spy command

### PR DESCRIPTION
**Pull request by Deku**
In the context of in-game player management, sometimes it's useful to know what abusive players are doing.

##### CHANGES PROPOSED:

-  Add a new spy command for game masters and administrators to receive all conversations of followed players.

##### TESTS PERFORMED:

- Build confirmed
- In-game use confirmed in my own server (production)
- Compilation done under Windows and Linux. In-game check on a Linux server.

##### HOW TO TEST THE CHANGES:

- Use command `.spy follow <characterName>` with a GM account
- Make that player whisper another player
- You (GM) should receive the message sent
- Use command `.spy status`. You should get a list with the test subject's name in it.
- Use command `.spy unfollow <characterName>`
- You should not receive any more messages sent or received by that player

##### KNOWN ISSUES AND TODO LIST:

None as for now

##### Target branch(es):

Master

### CODE REVIEW

File: data/sql/updates/db_world/2019_01_11_00.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+--
+-- START UPDATING QUERIES
+--
+
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1547165890320901800');
+
+DELETE FROM `trinity_string` WHERE `entry` BETWEEN 5062 AND 5071;
+INSERT INTO `trinity_string` (`entry`, `content_default`) VALUES
+(5062, 'You will now read %s conversations.'),
+(5063, 'Stopped spying %s.')
```
Comment (Winfidonarleyan) [inline data/sql/updates/db_world/2019_01_11_00.sql]:
```sql
(5063, 'Stopped spying %s.'),
```

Comment (Deku) [reply data/sql/updates/db_world/2019_01_11_00.sql]: Slipped right under my nose, thank you @Winfidonarleyan !

**Comment by BarbzYHOOL** [comment]
Awesome!

however there is a compilation error:

```
/home/travis/build/azerothcore/azerothcore-wotlk/src/server/scripts/Commands/cs_spy.cpp:227:74: fatal error: unused parameter 'args' [-Wunused-parameter]
    static bool HandleSpyStatusCommand(ChatHandler* handler, const char* args)
                                                                         ^
1 error generated.
src/server/scripts/CMakeFiles/scripts.dir/build.make:582: recipe for target 'src/server/scripts/CMakeFiles/scripts.dir/Commands/cs_spy.cpp.o' failed
```

File: src/server/scripts/Commands/cs_spy.cpp [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    static bool HandleSpyClearCommand(ChatHandler* handler, const char* /*args*/)
+    {
+        gmListening.clear();
+        handler->SendSysMessage(LANG_COMMAND_SPY_CLEARED);
+        return true;
+    }
+
+    static bool HandleSpyStatusCommand(ChatHandler* handler, const char* args)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_spy.cpp]:
```cpp
static bool HandleSpyStatusCommand(ChatHandler* handler, const char* /*args*/)
```

Comment (Deku) [reply src/server/scripts/Commands/cs_spy.cpp]: Seems like I'm using less strict compiler flags :sweat_smile:

Comment (BarbzYHOOL) [reply src/server/scripts/Commands/cs_spy.cpp]: Deku, do you edit it? (i don't edit anything you do else you get angry xD)

Comment (Deku) [reply src/server/scripts/Commands/cs_spy.cpp]: I did. Sorry, but I was busy 😬

**Pull request merged**